test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_copy_image.cpp

```diff
@@ -0,0 +1,500 @@
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandCopyImageKHR(command_buffer, queue, src_image,
+                                             dst_image, origin, origin, region,
+                                             0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_QUEUE,
+                               "clCommandCopyImageKHR should return "
+                               "CL_INVALID_COMMAND_QUEUE",
+                               TEST_FAIL);
+
+        error = clCommandCopyImageToBufferKHR(command_buffer, queue, src_image,
+                                              dst_image, origin, origin, 0, 0,
+                                              nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_QUEUE,
+                               "clCommandCopyImageKHR should return "
+                               "CL_INVALID_COMMAND_QUEUE",
+                               TEST_FAIL);
+
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        src_image = create_image_2d(context, CL_MEM_READ_ONLY, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        dst_image = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        return CL_SUCCESS;
+    }
+
+    clMemWrapper src_image;
+    clMemWrapper dst_image;
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { 512, 512, 1 };
+};
+
+// CL_INVALID_CONTEXT if the context associated with command_queue,
+// command_buffer, src_image, and dst_image are not the same.
+struct CommandBufferCopyImageContextNotSame : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandCopyImageKHR(
+            command_buffer, nullptr, src_image_ctx, dst_image, origin, origin,
+            region, 0, 0, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandCopyImageKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
+
+        error = clCommandCopyImageToBufferKHR(
+            command_buffer, nullptr, src_image_ctx, dst_image, origin, origin,
+            0, 0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_CONTEXT,
+                               "clCommandCopyImageKHR should return "
+                               "CL_INVALID_CONTEXT",
+                               TEST_FAIL);
```

Comment:
Minor - the same error occurs elsewhere in this file, also.

```suggestion
        test_failure_error_ret(error, CL_INVALID_CONTEXT,
                               "clCommandCopyImageToBufferKHR should return "
                               "CL_INVALID_CONTEXT",
                               TEST_FAIL);
```

Reply:
@bashbaug corrected